packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6267,46 +6266,77 @@ abstract class RenderObjectElement extends Element {
     return ancestor as RenderObjectElement?;
   }
 
-  ParentDataElement<ParentData>? _findAncestorParentDataElement() {
+  List<ParentDataElement<ParentData>> _findAncestorParentDataElements() {
+    // More than one ParentDataWidget can contribute ParentData, but there are
+    // some constraints.
+    // 1. ParentData can only be written by unique ParentDataWidgets.
```

Comment:
Maybe a little clearer?

```suggestion
    // 1. ParentData can only be written by unique ParentDataWidget types.
```